### PR TITLE
Add tests for partial_derivatives, sources, stress_intregrand, stress, & boundary.

### PR DESCRIPTION
**Pull request by icui**
Add tests for partial_derivatives, sources, stress_intregrand, stress, & boundary.

## Issue Number

Adds to #809 

## Checklist

Please make sure to check developer documentation on specfem docs.

- [ ] I ran the code through pre-commit to check style
- [ ] My code passes all the integration tests
- [ ] I have added sufficient unittests to test my changes
- [ ] I have added/updated documentation for the changes I am proposing
- [ ] I have updated CMakeLists to ensure my code builds
- [ ] My code builds across all platforms

### CODE REVIEW

File: tests/unit-tests/point/boundary_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+
+    // Set edge weight and normal
+    boundary.edge_weight = 2.5;
+    boundary.edge_normal(0) = 0.8;
+    boundary.edge_normal(1) = 0.6;
+
+    // Verify values
+    EXPECT_TRUE(boundary.tag == element::boundary_tag::none);
+    EXPECT_TRUE(specfem::datatype::all_of(
+        Kokkos::abs(boundary.edge_weight - 2.5) < tol))
```
Comment (lsawade) [inline tests/unit-tests/point/boundary_tests.cpp:141]: you could divide here by the value to get a relative error!

File: tests/unit-tests/point/partial_derivatives_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+  constexpr int simd_size =
+      specfem::datatype::simd<type_real, using_simd>::size();
+
+  // Value to use in constructor
+  typename specfem::datatype::simd<type_real, using_simd>::datatype const_val;
+
+  // Initialize value
+  if constexpr (using_simd) {
+    for (int i = 0; i < simd_size; ++i) {
+      const_val[i] = 7.7;
```
Comment (lsawade) [inline tests/unit-tests/point/partial_derivatives_tests.cpp:131]: for a const val you can call the simd_type constructor! so you don't need to loop here

File: tests/unit-tests/point/partial_derivatives_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+      specfem::datatype::simd<type_real, using_simd>::size();
+
+  // Values to use in constructor
+  typename specfem::datatype::simd<type_real, using_simd>::datatype xix_val;
+  typename specfem::datatype::simd<type_real, using_simd>::datatype gammax_val;
+  typename specfem::datatype::simd<type_real, using_simd>::datatype xiz_val;
+  typename specfem::datatype::simd<type_real, using_simd>::datatype gammaz_val;
+
+  // Initialize values
+  if constexpr (using_simd) {
```
Comment (lsawade) [inline tests/unit-tests/point/partial_derivatives_tests.cpp:167]: same here

**Review by lsawade — APPROVED** [review]
LGTM, minor comments
